src/ext/coms_curl.c

```diff
@@ -251,6 +253,8 @@ static void *writer_loop(void *_) {
         signal_data_processed(writer);
     } while (running);
 
+    curl_slist_free_all(writer->headers);
```

Comment:
Good catch. Was pretty sure this was already handled. Must have been lost in one of WIP PRs.
Thanks for fixing this!